### PR TITLE
JSDK-2456 port dscp fix + RTCRtpSender.setParameters use.

### PR DESCRIPTION
**Pull request by makarandp0**
This change cherry-picks and merges following PRs that were already merged into 1.x branch
 
https://github.com/twilio/twilio-video.js/pull/707 Fixing failing DSCP tagging test. 
https://github.com/twilio/twilio-video.js/pull/704 JSDK-2440 support dscp tagging (support-1.x-chrome-planb) 
https://github.com/twilio/twilio-video.js/pull/700 JSDK-2250 Use RTCRtpSender.setParameters where available, instead of SDP munging. 

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,23 @@ For 1.x changes, go [here](https://github.com/twilio/twilio-video.js/blob/suppor
...
 
+New Features
+------------
+
+- You can now enable [dscp tagging](https://tools.ietf.org/html/draft-ietf-tsvwg-rtcweb-qos-18) for audio
+  packets by specifying new ConnectOptions property `dscpTagging` and setting it to `true`.
+  DSCP tagging allows you to request enhanced QoS treatment for audio packets from any firewall/routers
+  that support this feature. Setting this option to `true` will request DSCP tagging
+  for audio packets on supported browsers (only Chrome supports this as of now). Audio packets will be
+  sent with DSCP header value set to (0xb8) which corrosponds to EF = Expediated Forwarding.
```
Comment (manjeshbhargav) [inline CHANGELOG.md]: @makarandp0 `s/Expediated/Expedited/g`.

Comment (manjeshbhargav) [reply CHANGELOG.md]: @makarandp0 Also, please add JIRA ticket.

File: lib/connect.js [hunk 2]
```diff
@@ -289,6 +290,11 @@ function connect(token, options) {
  *   flag does not have any effect in a Peer-to-Peer Room.
  * @property {boolean} [dominantSpeaker=false] - Whether to enable the Dominant
  *   Speaker API or not. This only takes effect in Group Rooms.
+ * @property {boolean} [dscpTagging=false] - DSCP tagging allows you to request enhanced
+ * QoS treatment for RTP media packets from any firewall that the client may be behind.
+ * Setting this option to <code>true</code> will request DSCP tagging for audio packets
+ * on supported browsers (only Chrome supports this as of now). Audio packets will be
+ * sent with DSCP header value set to (0xb8) which corrosponds to EF = Expediated Forwarding.
```
Comment (manjeshbhargav) [inline lib/connect.js]: @makarandp0 `s/Expediated/Expedited/g`.

File: lib/signaling/v2/peerconnection.js [hunk 3]
```diff
@@ -1317,4 +1353,50 @@ function updateRecycledTransceivers(pcv2) {
   });
 }
 
+/**
+ * Perform certain updates after an SDP negotiation is completed.
+ * @param {PeerConnectionV2} pcv2
+ * @returns {void}
+ */
+function negotiationCompleted(pcv2) {
```
Comment (manjeshbhargav) [inline lib/signaling/v2/peerconnection.js]: @makarandp0 Can we enhance this function to:
```js
function negotiationCompleted(pcv2) {
  if (pcv2._isUnifiedPlan) {
    updateRecycledTransceivers(this);
    updateLocalCodecs(this);
    updateRemoteCodecMaps(this);
  }
  if (pcv2._isRTCRtpSenderParamsSupported) {
    updateEncodingParameters(pcv2);
  }
}
```
?

File: lib/signaling/v2/peerconnection.js [hunk 4]
```diff
@@ -817,6 +845,7 @@ class PeerConnectionV2 extends StateMachine {
             updateLocalCodecs(this);
             updateRemoteCodecMaps(this);
           }
+          negotiationCompleted(this);
```
Comment (manjeshbhargav) [inline lib/signaling/v2/peerconnection.js]: @makarandp0 If we enhance `negotiationCompleted` like I've suggested then, this will simplify to:
```js
...
else if (description.type === 'answer') {
  this._lastStableDescriptionRevision = this._descriptionRevision;
  negotiationCompleted(this);
}
```

File: lib/signaling/v2/peerconnection.js [hunk 5]
```diff
@@ -865,10 +896,14 @@ class PeerConnectionV2 extends StateMachine {
...
-        this._isRestartingIce = false;
+      if (description.type === 'answer') {
+        if (this._isRestartingIce) {
+          this._log.debug('An ICE restart was in-progress and is now completed');
+          this._isRestartingIce = false;
+        }
+        negotiationCompleted(this);
       }
+
       if (description.type === 'answer' && this._isUnifiedPlan) {
```
Comment (manjeshbhargav) [inline lib/signaling/v2/peerconnection.js]: @makarandp0 If we enhance `negotiationCompleted` like I've suggested then, we can remove this `if` block.

File: lib/signaling/v2/peerconnection.js [hunk 6]
```diff
@@ -933,7 +968,8 @@ class PeerConnectionV2 extends StateMachine {
     const revision = description.revision;
     return Promise.resolve().then(() => {
       return this._setRemoteDescription(description);
-    }).catch(() => {
+    }).catch((error) => {
+      console.error(error.message);
```
Comment (manjeshbhargav) [inline lib/signaling/v2/peerconnection.js]: @makarandp0 Revert this change after debugging.

File: lib/signaling/v2/peerconnection.js (hunk 3 above)
Comment (makarandp0) [reply lib/signaling/v2/peerconnection.js]: thanks @manjeshbhargav, That looks much better.

**Comment by makarandp0** [comment]
Thanks @manjeshbhargav -  I have incorporated your feedback, and this is ready for another look when you get a chance.

File: test/integration/spec/connect.js [hunk 7]
```diff
@@ -504,7 +515,50 @@ describe('connect', function() {
...
+        });
+
+        after(() => {
+          [thisRoom, ...thoseRooms].forEach(room => room.disconnect());
+        });
+      });
+    });
+  });
+
+  describe.only('called with EncodingParameters', () => {
```
Comment (manjeshbhargav) [inline test/integration/spec/connect.js]: @makarandp0 Please remove `.only(`.

File: lib/signaling/v2/peerconnection.js [hunk 8]
```diff
@@ -1317,4 +1341,56 @@ function updateRecycledTransceivers(pcv2) {
...
+ * @param {PeerConnectionV2} pcv2
+ * @returns {void}
+ */
+function negotiationCompleted(pcv2) {
+  if (pcv2._isUnifiedPlan) {
+    updateRecycledTransceivers(pcv2);
+    updateLocalCodecs(pcv2);
+    updateRemoteCodecMaps(pcv2);
+  }
+
```
Comment (manjeshbhargav) [inline lib/signaling/v2/peerconnection.js]: @makarandp0 Nitpick: remove extra line.

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,23 @@ For 1.x changes, go [here](https://github.com/twilio/twilio-video.js/blob/suppor
 2.0.0-beta13 (in progress)
 ==========================
 
+New Features
+------------
+
+- You can now enable [dscp tagging](https://tools.ietf.org/html/draft-ietf-tsvwg-rtcweb-qos-18) for audio
```
Comment (manjeshbhargav) [inline CHANGELOG.md]: @makarandp0 Nitpick: `s/dscp/DSCP`?
